### ISSUE TITLE
Can't find ISO8601Date

### ISSUE THREAD

**Issue by connorshea** [rmosolgo/graphql-ruby#2527]
Hi! I just upgraded to graphql 1.9.13 and I can't figure out how to get the new ISO8601Date type (from #2471) to work.

I tried adding it to my `GameType` class (simplified for this issue), like this:

```ruby
module Types
  class GameType < Types::BaseObject
    field :id, ID, null: false
    field :name, String, null: false
    field :description, String, null: true
    field :release_date, GraphQL::Types::ISO8601Date, null: true
  end
end
```

But this results in my API request specs failing with one of two errors:

```
1) Series API Query for data on series returns data for a series when searching
     Failure/Error: field :release_date, GraphQL::Types::ISO8601Date, null: true
     
     NameError:
       uninitialized constant GraphQL::Types::ISO8601Date
       Did you mean?  GraphQL::Types::ISO8601DateTime
     # ./app/graphql/types/game_type.rb:12:in `<class:GameType>'
     # ./app/graphql/types/game_type.rb:3:in `<module:Types>'
     # ./app/graphql/types/game_type.rb:2:in `<top (required)>'
     # ./app/graphql/types/query_type.rb:4:in `<class:QueryType>'
     # ./app/graphql/types/query_type.rb:3:in `<module:Types>'
     # ./app/graphql/types/query_type.rb:2:in `<top (required)>'
     # ./app/graphql/video_game_list_schema.rb:11:in `<class:VideoGameListSchema>'
     # ./app/graphql/video_game_list_schema.rb:2:in `<top (required)>'
     # ./spec/requests/api/series_spec.rb:46:in `block (3 levels) in <top (required)>'
```

```
  8) Games API Query for data on games returns cover for game
     Failure/Error:
       result = VideoGameListSchema.execute(
         query_string,
         context: { current_user: user },
         variables: { id: game_with_cover.id }
       )
     
     RuntimeError:
       Unexpected parent_type: 
     # ./spec/requests/api/games_spec.rb:48:in `block (3 levels) in <top (required)>'
```

And causes the actual API requests to fail as well.

When I use `bundle open graphql`, the `lib/graphql/types/iso_8601_date.rb` file definitely exists.

When I look at GraphiQL (without the changes that add a `release_date` field, since GraphiQL errors if I try to use the ISO8601Date class), I don't see any reference to an `ISO8601Date` type. Only the DateTime type. (Though, I also don't see the JSON type, so maybe it's just because I'm not using it?)

<img width="334" alt="Screen Shot 2019-10-08 at 10 08 11 PM" src="https://user-images.githubusercontent.com/2977353/66451081-24430f00-ea18-11e9-8783-6c608e207cf0.png">

I'm using Sorbet, but I'm fairly certain that's not relevant to the problem. I re-ran  `srb init` to try and get it to detect the new class, but Sorbet didn't find it, and it wasn't added to the graphql gem's RBI file. Maybe it's not being autoloaded for some reason?

I've pushed a commit up to my repository with the broken code, if you want to try running it: https://github.com/connorshea/VideoGameList/commit/f7c9fdf87dd763cf1fd0456f8e11cf71de41fbb7

The instructions in the project's readme should be sufficient, but honestly it's probably too much work to get it all set up just for this.

If you need any more information from me, I can try to provide it.

Ruby version: 2.6.5
Rails version: 6.0.0
graphql-ruby version: 1.9.13
OS: macOS Mojave 10.14.6

**Comment by connorshea** [comment]
Ah, is it because the date class isn't required in [`lib/graphql/types.rb`](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/types.rb)?

**Comment by connorshea** [comment]
I've confirmed that fixes it for me, and opened #2528 👍

**Comment by rmosolgo** [comment]
Thanks for your detailed report here, and for pushing a fix! It'll be on rubygems soon, in the meantime, you can bundle master with 

```
gem "graphql", github: "rmosolgo/graphql-ruby", ref: "master"
```
